packages/flutter/lib/src/material/input.dart

```diff
@@ -13,6 +13,126 @@ import 'theme.dart';
 
 export 'package:sky_services/editing/editing.mojom.dart' show KeyboardType;
 
+/// A container for grouping together multiple Input widgets.
+class Form extends StatefulWidget {
+  Form({
+    Key key,
+    this.child,
+    this.data,
+    this.onSubmitted
+  }) : super(key: key) {
+    assert(child != null);
+  }
+
+  /// A map of form field IDs to their input values.
+  final Map<String, String> data;
+
+  /// Called when the input is accepted anywhere on the form.
+  final VoidCallback onSubmitted;
```

Comment:
I'm open to adding that param. I'm a little confused about the desired onSubmitted behavior, though. Currently it fires when editing any input field and hitting the `enter` key. Should that indicate that the form be submitted? Should it matter which field was focused? The callback does not fire when changing focus between fields. To me, that indicates that hitting `enter` should submit the whole form.

Thoughts?